risc0/zkvm/src/exec/mod.rs

```diff
@@ -0,0 +1,575 @@
+// Copyright 2023 RISC Zero, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! TODO
+
+mod io;
+#[cfg(test)]
+mod tests;
+
+use std::{
+    array,
+    collections::{BTreeSet, HashMap},
+};
+
+use anyhow::{anyhow, bail, Result};
+use builder_pattern::Builder;
+use risc0_zkp::{
+    core::{
+        digest::{DIGEST_BYTES, DIGEST_WORDS},
+        hash::sha::BLOCK_BYTES,
+    },
+    ZK_CYCLES,
+};
+use risc0_zkvm_platform::{
+    memory::SYSTEM,
+    syscall::{
+        ecall, halt,
+        reg_abi::{REG_A0, REG_A1, REG_A2, REG_A3, REG_A4, REG_T0},
+    },
+    PAGE_SIZE, WORD_SIZE,
+};
+use rrs_lib::{
+    instruction_executor::InstructionExecutor,
+    instruction_string_outputter::InstructionStringOutputter, process_instruction, HartState,
+    MemAccessSize, Memory,
+};
+
+use self::io::{Syscall, SyscallContext};
+use crate::{align_up, binfmt::image::PageTableInfo, Loader, Segment};
+use crate::{
+    opcode::{MajorType, OpCode},
+    ExitCode, MemoryImage, Session,
+};
+
+const DEFAULT_SEGMENT_LIMIT_PO2: usize = 20; // 1M cycles
+
+const DEFAULT_SESSION_LIMIT: usize = 64 * 1024 * 1024; // 64M cycles
+
+// The number of cycles required to compress a SHA-256 block.
```

Comment:
I think these should be doc comments (`///`) so `--document-private-items` of rustdoc will pick them up.

Reply:
I'm trying hard to shrink the public interface to reduce the burden of ensuring all public interfaces have docs. Do you use a local rustdoc build with `--document-private-items`?